plugins/versioning/README.md

```diff
@@ -0,0 +1,332 @@
+
+```
+
+Alternatively, via JSON configuration:
+```json
+{
+  "moduleName": "Dokka Example",
+  ...
+  "pluginsConfiguration": [
+    {
+      "fqPluginName": "org.jetbrains.dokka.versioning.VersioningPlugin",
+      "serializationFormat": "JSON",
+      "values": "{\"version\": \"1.5\", \"versionsOrdering\": [\"1.5\", \"1.4\", \"1.3\", \"1.2\", \"1.1\", \"alpha-2\", \"alpha-1\"], \"olderVersionsDir\": \"documentation/version\", \"olderVersions\": [\"documentation/alpha/alpha-2\", \"documentation/alpha/alpha-1\"], \"renderVersionsNavigationOnAllPages\": true}"
+    }
+  ]
+}
+```
+
+</details>
+
+## Generating versioned documentation
+
+With the versioning plugin applied and configured, no other steps are needed. Documentation can be built in the usual way.
+
+Among other things, the versioning plugin adds a `version.json` file to the output folder. This file is used by the
+plugin to match versions and generate version navigation. If your previously generated documentation does not have that
+file, you will need to re-generate documentation for such versions. Just adding the file will not work.
+
+The versioning plugin also bundles all other documentation versions that have been passed through `olderVersionsDir`
+and `olderVersions` configuration options by putting them inside the `older` directory.
+
+## Usage example
+
+There is no single correct way to configure the plugin, it can be tailored to your needs. However,
+it can be a bit overwhelming when starting out. Below you will find one of the ways it can be configured so that you
+can begin publishing versioned documentation straight away.
+
+The main idea behind it is the following:
+
+1. One directory contains all versions of your documentation. For example, `documentation/version/{doc_version}`.
+   This is your archive which is needed for future builds.
+2. The output directory of all new builds is set to that directory as well, under `documentation/version/{new_version}`.
+3. When new builds are executed, the plugin looks for previous versions of documentation in the archive directory.
+4. Once new documentation has been generated, it needs to be **copied** to somewhere accessible by the user.
+   For example, GitHub pages or nginx static directories. It needs to be **copied**, not moved because Dokka will still
+   need this version for future builds, otherwise there will be a gap in the archive.
+5. Once it has been safely copied, you can remove the `older` directory from the newly generated and archived version.
+   This helps reduce the overhead of each version bundling all previous versions, as these files are effectively duplicates.
+
+```kotlin
+import org.jetbrains.dokka.versioning.VersioningPlugin
+import org.jetbrains.dokka.versioning.VersioningConfiguration
+
+buildscript {
+    dependencies {
+        classpath("org.jetbrains.dokka:versioning-plugin:1.7.20")
+    }
+}
+
+dependencies {
+    dokkaPlugin("org.jetbrains.dokka:versioning-plugin:$dokkaVersion")
```

Comment:
Probably, `$dokkaVersion` should be replaced with the exact version as everywhere

Reply:
Yep, you're right! Fixed